content/en/docs/concepts/storage/volumes.md

```diff
@@ -1213,7 +1213,75 @@ in `containers[*].volumeMounts`. Its values are:
   (unmounted) by the containers on termination.
   {{< /warning >}}
 
+## Read-only mounts
 
+A mount can be made read-only by setting the `.spec.containers[].volumeMounts[].readOnly`
+field to `true`.
+
+On Linux, read-only mounts are not recursively read-only by default.
+For example, if you have another mount such as tmpfs, NFS, or USB storage
+file system under `/mnt/<SUBMOUNT>`, even if you specify to mount `/mnt` into a
+container as read-only, `/mnt/<SUBMOUNT>` will become still writable.
+
+### Recursive read-only mounts
+
+{{< feature-state for_k8s_version="v1.30" state="alpha" >}}
+
+Recursive read-only mounts can be enabled by setting the
+`RecursiveReadOnlyMounts` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+for kubelet and kube-apiserver, and setting the `.spec.containers[].volumeMounts[].recursiveReadOnly`
+field for a pod.
+
+The allowed values are:
+
+* `Disabled` (default): no effect.
+
+* `Enabled`: makes the mount recursively read-only.
+  Needs all the following requirements to be satisfied:
+  * `readOnly` is set to `true`
+  * `mountPropagation` is unset, or, set to `None`
+  * The host is running with Linux kernel v5.12 or later
+  * The runtime class supports recursive read-only mounts.
+    e.g., runc v1.1 or later.
+
+* `IfPossible`: attempts to apply `Enabled`, and falls back to `Disabled`
+  if the feature is not supported by the kernel or the runtime class.
+
+Example:
```

Comment:
(nit)

Ideally (and we'll insist on this for beta): use a [downloadable code sample](https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#source-code-files). We also test that these samples validate, whereas we don't do that for YAML snippets in pages.

Reply:
done